File: despatch.WEB/app/src/swagger/models/room-response.ts
/* tslint:disable */
import { BaseEntityResponse } from './base-entity-response';
import { ContactResponse } from './contact-response';
import { UserResponse } from './user-response';
import { MessageResponse } from './message-response';
import { RoomPrivacyType } from './room-privacy-type';
import { RoomImageResponse } from './room-image-response';
import { RoomType } from './room-type';
import { UserRoomResponse } from './user-room-response';
export interface RoomResponse extends BaseEntityResponse{
  contacts?: Array<ContactResponse>;
  creator?: UserResponse;
  creatorId?: string;
  description?: string;
  lastMessage?: MessageResponse;
  lastMessageId?: string;
  name?: string;
  privacyType: RoomPrivacyType;
  roomImages?: Array<RoomImageResponse>;
  type: RoomType;
  userRooms?: Array<UserRoomResponse>;
}

File: despatch.Domain/Models/Default/Response/RoomResponse.cs
﻿using despatch.Domain.Models.Enums;
using System.Collections.Generic;

namespace despatch.Domain.Models.Default.Response
{
    public class RoomResponse : BaseEntityResponse
    {
        public string Name { get; set; }
        public string Description { get; set; }
        public RoomType Type { get; set; }
        public RoomPrivacyType PrivacyType { get; set; }

        public string CreatorId { get; set; }
        public string LastMessageId { get; set; }

        public UserResponse Creator { get; set; }
        public MessageResponse LastMessage { get; set; }
        public List<ContactResponse> Contacts { get; set; }

        public virtual List<UserRoomResponse> UserRooms { get; set; }
        public virtual List<RoomImageResponse> RoomImages { get; set; }
    }
}
